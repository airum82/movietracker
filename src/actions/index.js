export const addNowPlaying = (nowPlaying) => ({
  type: 'ADD_NOW_PLAYING',
  nowPlaying
})

export const createAccount = (userName, password) => ({
  type: 'CREATE_ACCOUNT',
  userName,
  password
})