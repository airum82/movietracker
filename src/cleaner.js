export const recentMovies = (nowPlaying) => {
  return nowPlaying.map(movie => {
    return {
      title: movie.title,
      overview: movie.overview,
      poster: movie.poster_path,
      ratings: movie.vote_average,
      id: movie.id,
      releaseDate: movie.release_date,
      favorite: false
    };
  });
};

export const makeFavoriteMovie = (movie, userId) => {
  const { title, overview, poster, ratings, id, releaseDate } = movie;
  return { 
    movie_id: id, 
    user_id: userId, 
    title, 
    poster_path: poster, 
    release_date: releaseDate,
    overview,
    vote_average: ratings,
  }
}

export const addFavoriteKey = (movies) => {
  return movies.map(movie => {
    return {...movie, favorite: true}
  })
}

export const checkForFavorites = (nowPlaying, favorites = []) => {
  const newNowPlaying = nowPlaying.map(movie => {
    const foundMovie = favorites.find(favorite => favorite.movie_id == movie.id)
    if(foundMovie) {
      return {...movie, favorite: true}
    }
      return movie
  })
  return newNowPlaying;
}

export const toggleFavorite = (nowPlaying, id) => {
  return nowPlaying.map(movie => {
    if(movie.id === id) {
      return {...movie, favorite: !movie.favorite}
    }
    return movie
  })
}

export const updateFavorites = (nowPlaying) => {
  return nowPlaying.filter(movie => movie.favorite)
}