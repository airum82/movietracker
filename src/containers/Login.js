import React, {Component} from 'react'
import {connect} from 'react-redux'

export class Login extends Component {
  constructor() {
    super()
    this.state = {
      userName: '',
      password: ''
    }
  }

  handleChange = (e) => {
    const {value, name} = e.target
    this.setState({
      [name]: value
    })
  }

  submitAccount = (e) => {
    e.preventDefault()
    this.props.handleSubmit(this.state.userName, this.state.password)
  }

  render() {
    return (
      <div>
        <form onSubmit={this.submitAccount}>
          <input 
            type="text" 
            value={this.state.userName}
            onChange={this.handleChange}
            name='userName'
          />
          <input 
            type="text" 
            value={this.state.password}
            onChange={this.handleChange}
            name='password'
          />
        </form>
      </div>
    )
  }
}
