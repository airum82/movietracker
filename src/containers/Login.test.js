import React from 'react';
import { Login, mapDispatchToProps } from './Login';
import { shallow, mount } from 'enzyme';
import { logIn } from '../actions';
import { cleanFavorites } from '../cleaner';
import { NavLink, BrowserRouter } from 'react-router-dom';

describe('Login', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Login handleSubmit={jest.fn()} />);
  });

  it('should match snapshot upon render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange on username input change', () => {
    wrapper = mount(
      <BrowserRouter>
        <Login />
      </BrowserRouter>);
    const spy = jest.spyOn(wrapper.find('Login').instance(), 'handleChange');
    const mockEvent = { target: { 
      name: 'username',
      value: 'a'} };
    wrapper.find('Login').instance().handleChange(mockEvent);
    wrapper.find('.username').simulate('change');
    expect(spy).toHaveBeenCalled();
  });

  it('should call handleChange on password input change', () => {
    wrapper = mount(
      <BrowserRouter>
        <Login />
      </BrowserRouter>);
    const spy = jest.spyOn(wrapper.find('Login').instance(), 'handleChange');
    const mockEvent = { target: { 
      name: 'password',
      value: 'a'} };
    wrapper.find('Login').instance().handleChange(mockEvent);
    wrapper.find('.password').simulate('change');
    expect(spy).toHaveBeenCalled();
  });

  it.skip('should call submitAccount on submit', () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({ data: { id: 6 }})
    }));
    const spy = jest.spyOn(wrapper.instance(), 'submitAccount');
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().submitAccount(mockEvent);
    wrapper.simulate('submit');
    expect(spy).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch when using a function from MDTP', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = {email: 'michael', password: 'password'};
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.handleSubmit(actionToDispatch);
      expect(mockDispatch).toHaveBeenCalledWith(logIn(actionToDispatch));
    });
  });
});