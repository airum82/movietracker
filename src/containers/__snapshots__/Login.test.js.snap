// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="login-form"
          onSubmit={[Function]}
>
          <input
                    className="username"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value=""
          />
          <input
                    className="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <button
                    className="submit-button"
          >
                    Log in
          </button>
</form>,
        <div
          className="create-account-div"
>
          <p
                    className="error-message"
          />
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="create-account"
                    to="/createAccount"
          >
                    or Create Account
          </NavLink>
</div>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="username"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
/>,
            <input
              className="password"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <button
              className="submit-button"
>
              Log in
</button>,
          ],
          "className": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "username",
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log in",
              "className": "submit-button",
            },
            "ref": null,
            "rendered": "Log in",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="error-message"
/>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="create-account"
              to="/createAccount"
>
              or Create Account
</NavLink>,
          ],
          "className": "create-account-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "error-message",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "or Create Account",
              "className": "create-account",
              "to": "/createAccount",
            },
            "ref": null,
            "rendered": "or Create Account",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="login-form"
            onSubmit={[Function]}
>
            <input
                        className="username"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
            />
            <input
                        className="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
            <button
                        className="submit-button"
            >
                        Log in
            </button>
</form>,
          <div
            className="create-account-div"
>
            <p
                        className="error-message"
            />
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="create-account"
                        to="/createAccount"
            >
                        or Create Account
            </NavLink>
</div>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="username"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
/>,
              <input
                className="password"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
              <button
                className="submit-button"
>
                Log in
</button>,
            ],
            "className": "login-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "username",
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log in",
                "className": "submit-button",
              },
              "ref": null,
              "rendered": "Log in",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="error-message"
/>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="create-account"
                to="/createAccount"
>
                or Create Account
</NavLink>,
            ],
            "className": "create-account-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "error-message",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "or Create Account",
                "className": "create-account",
                "to": "/createAccount",
              },
              "ref": null,
              "rendered": "or Create Account",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
